board/safety/safety_volkswagen.h

```diff
@@ -0,0 +1,169 @@
+const int VW_MAX_STEER = 300;               // 3.0 nm
+const int VW_MAX_RT_DELTA = 130;            // max delta torque allowed for real time checks
+const uint32_t VW_RT_INTERVAL = 250000;     // 250ms between real time checks
+const int VW_MAX_RATE_UP = 10;
```

Comment:
you might get better performance (for the safety) by having the rate up lower (in absolute value) than the rate down. From a safety perspective, there is really no constraint on the rate down number, since having an EPS fault would cause an immediate loss of torque anyway. 

Is the steer command sent at 100Hz?

Reply:
The rack will tolerate a range of send rates. Stock sends at 50Hz when steering and 10Hz when not steering. We have tested sending at 100Hz and it works fine, but I did not perceive any real benefit and reverted to 50Hz.

The rack seems to tolerate literally anything for up/down rate; during development I let OP send through unlimited raw values without OP or Panda limiting in place. However, we had some really tough tuning problems, nasty high amplitude oscillations at low speed and so forth. I recently got information from "a source" that the _effective_ rate may be shaped to a 5nm/sec rate of change. It's not clear if that applies to both up and down.

Reply:
Ok, we can push tuning for later and keep this "as is" for now.